R/intensity.R

```diff
@@ -18,32 +18,54 @@ get_british_ci <-
     
     if (all(is.null(c(start, end)))) {
       call <- url
+      result <- query_british_ci_api(call)
+      return(result)
     } else if (all(!is.null(c(start, end)))) {
-      from_date <- paste0(as.Date(start), 'T00:00Z/')
-      to_date <- paste0(as.Date(end), 'T23:59Z')
-      
-      call <- paste0(url, from_date, to_date)
+      # No error if both specified
     } else {
       stop('Both start and end arguments have to be either NULL or in use')
     }
     
-    data <- get_data(call)
+    # List all the dates out
+    dates <- seq(lubridate::ymd(start), lubridate::ymd(end), by = '1 day')
     
-    if (is.list(data) && length(data) == 0) {
-      result <- NULL
-    } else {
-      result <- data %>%
-        dplyr::mutate(from = lubridate::ymd_hm(.data$from),
-                      to = lubridate::ymd_hm(.data$to)) %>%
-        tibble::as_tibble()
-      
-      result <- clean_colnames(result)
-    }
+    # Split the list into maximum of 14 days to prevent api errors
+    result <- purrr::map_df(split(dates, ceiling(seq_along(dates) / 14)), function(i){
+      start = min(i)
+      end = max(i)
+      from_date <- paste0(as.Date(start), 'T00:00Z/')
+      to_date <- paste0(as.Date(end), 'T23:59Z')
+      call <- paste0(url, from_date, to_date)
+      result <- query_british_ci_api(call)
+    })
```

Comment:
Great implementation of `ceiling()` and `map` functionality + I totally agree with your approach of wrapping lower-level API call in `query_british_ci_api()`! One thing I'd like to add is that this code chunk belongs to the `else if()` part of the for loop so it would be nice to move it there for better code readability.